pkgs/build-support/setup-hooks/wrap-gapps-hook.sh

```diff
@@ -36,16 +36,38 @@ wrapGAppsHook() {
   done
 
   if [[ -z "$dontWrapGApps" ]]; then
+    targetDirsThatExist=()
+    targetDirsRealPath=()
+
+    # wrap binaries
     targetDirs=( "${prefix}/bin" "${prefix}/libexec" )
     for targetDir in "${targetDirs[@]}"; do
       if [[ -d "${targetDir}" ]]; then
-        find -L "${targetDir}" -type f -executable -print0 \
+        targetDirsThatExist+=("${targetDir}")
+        targetDirsRealPath+=("$(realpath "${targetDir}")/")
+        find "${targetDir}" -type f -executable -print0 \
           | while IFS= read -r -d '' file; do
-          echo "Wrapping program ${file}"
+          echo "Wrapping program '${file}'"
           wrapProgram "${file}" "${gappsWrapperArgs[@]}"
         done
       fi
     done
+
+    # wrap links to binaries that point outside targetDirs
+    # Note: links to binaries within targetDirs do not need
+    #       to be wrapped as the binaries have already been wrapped
+    find "${targetDirsThatExist[@]}" -type l -xtype f -executable -print0 \
+      | while IFS= read -r -d '' linkPath; do
+        linkPathReal=$(realpath "${linkPath}")
+        for targetPath in "${targetDirsRealPath[@]}"; do
```

Comment:
Making targetDirsRealPath an associative array would be preferable here to avoid
quadratic complexity.

```bash
if [[ -n ${targetDirsRealPath[$linkpath]} ]]; then
   echo "Not wrapping link: '$linkPath' (already wrapped)"
   continue
fi
```

Reply:
The alternative would be to memoize each wrapped programs once.

Reply:
@Mic92 Thanks for the review and for the nix-review program which proved very helpful in testing.

I'm not sure if an associative array will help here as the comparison at line 63 is intended to match any linkPathReal that starts with the prefix targetPath:

`if [[ "$linkPathReal" == "$targetPath"* ]]; then`

In the context of this script, targetDirsRealPath should have at most two elements, so performance should remain linear with respect to iterating over the symbolic links.

That said, your comment prompted me to review the code again and I think that there is a bug if neither /bin or /libexec exists in a package. I'll need to fix that!

Reply:
@Mic92 Thanks, again, for the review. I made the fix for when neither /bin or /libexec are present. I also switched to the staging branch as per @FRidh with the assistance of @grahamc.